Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift

```diff
@@ -5166,12 +5138,10 @@ class IRRootFieldBuilderTests: XCTestCase {
       )
     ))
 
-    expect(allAnimals_asDog_deferredAsRoot).to(beNil())
+    expect(allAnimals_asDog?.containsDeferredChildFragment).to(beFalse())
```

Comment:
Testing for nil didn't mean there wasn't any deferred child, just that you didn't find it at the type info you tried it on. This new check better ensures no deferred children in the tree.